utils/parse/parse.go

```diff
@@ -8,16 +8,17 @@ import (
 
 // Port returns the port from an address
 func Port(address string) (uint, error) {
-	parts := strings.SplitN(address, ":", 2)
-	if len(parts) != 2 {
+	parts := strings.Split(address, ":")
```

Comment:
We have a builtin `url.Parse()` for this. This gives a `Host` field which you can safely check on presence of a port (format is always `host:port`)